### ISSUE TITLE
Quarkus maven plugin creates projects with non-relocated dependencies

### ISSUE THREAD

**Issue by fedinskiy** [quarkusio/quarkus#39330]
### Describe the bug

I am using quarkus-maven-plugin to create a project. When I add `resteasy-client` dependency, the resulting project gets `quarkus-rest-client-jackson` depedendency in pom.xml. When this project is compiled, there is a warning, that this dependency was relocated to `quarkus-resteasy-client-jackson`

### Expected behavior

Generated project can be compiled without warnings

### Actual behavior

```
[WARNING] The artifact io.quarkus:quarkus-rest-client-jackson:jar:3.8.2 has been relocated to io.quarkus:quarkus-resteasy-client-jackson:jar:3.8.2: io.quarkus:quarkus-rest-client-jackson:3.8.2 was relocated to io.quarkus:quarkus-resteasy-client-jackson:3.8.2. Please update the artifactId of the dependency in your project configuration. For more information about this change, please refer to https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.7#rest-client

````

### How to Reproduce?

1. generate project :`mvn io.quarkus.platform:quarkus-maven-plugin:3.8.2:create -DprojectGroupId=my-groupId -DprojectArtifactId=app-generated-skeleton   -DquarkusRegistryClient=false -DplatformArtifactId=quarkus-bom -DplatformVersion=3.8.2 -Dextensions=resteasy-client`
2. Compile the project: `mvn clean compile -f app-generated-skeleton/pom.xml`

### Output of `uname -a` or `ver`

6.7.3-200.fc39.x86_64

### Output of `java -version`

Java version: 21.0.1, vendor: Eclipse Adoptium

### Quarkus version or git rev

3.8.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)

**Comment by gsmet** [comment]
Good catch, on it.

**Comment by gsmet** [comment]
https://github.com/quarkusio/quarkus/pull/39334 should fix it for 3.8, I'll work on something for main too as it's probably a good idea to add a rest-client quickstart for the one from RESTEasy Reactive.
